pysat/_instrument.py

```diff
@@ -1873,19 +1873,12 @@ def bounds(self, value=None):
                                          'order.', istart, 'occurs after',
                                          istop))
                         raise ValueError(estr)
-                    itemp = self.files.get_file_array([istart], [istop])
-                    # downselect based upon step size
+                    # Account for width of load. Don't extend past bound.
+                    stop_idx = stop_idx - self._iter_width + 1
+                    # Stop index is exclusive when called this way, pad by 1
+                    itemp = self.files.files.values[start_idx:(stop_idx + 1)]
```

Comment:
Rewrote based on the functionality below.  Note that extra padding is needed since calling by indices has an exclusive endpoint, whereas the date function below has an inclusive endpoint.  tests are working locally.